packages/phone-number-privacy/signer/src/pnp/endpoints/sign/action.ts

```diff
@@ -100,30 +106,28 @@ export class PnpSignAction implements Action<SignMessageRequest> {
         name: DefaultKeyName.PHONE_NUMBER_PRIVACY,
       }
 
-      // Compute signature inside transaction so it will rollback on error.
-
       try {
         const signature = await this.sign(
           session.request.body.blindedQueryPhoneNumber,
           key,
           session
         )
         this.io.sendSuccess(200, session.response, key, signature, quotaStatus, session.errors)
+        return
       } catch (err) {
+        session.logger.error({ err })
+        // Note that errors thrown after rollback will have no effect
```

Comment:
Out of curiousity, what do you think would happen if we moved the rollback to line 130 after the response is sent but before we return? This might be slightly better on the off chance the response sending function throws (though it shouldn't)

Reply:
Ugh yeah I went back and forth a bit on this but not totally sure which way is better :-/ I don't love decrementing performedQueryCount before we know the rollback succeeded and figured that if sendFailure fails here, it would likely fail in the outer handler as well (maybe an unfair assumption), but maybe possible inconsistencies in received responses is better than not handling possible sendFailure errors 🤔 Happy to switch for now and we can keep this in mind